spec/solver_spec.rb

```diff
@@ -0,0 +1,37 @@
+require_relative '../solver'
+
+describe Solver do
+  before :each do
+    @solver = Solver.new
+  end
+
+  context 'when factorial method is called' do
+    it 'returns the factorial of a given number' do
+      expect(@solver.factorial(0)).to eq(1)
+      expect(@solver.factorial(1)).to eq(1)
+      expect(@solver.factorial(4)).to eq(24)
+      expect(@solver.factorial(5)).to eq(120)
+      expect(@solver.factorial(-1)).to eq('Invalid input')
```

Comment:
- According to the requirement, you should handle negative value which raises an exception as an expectation i.e instead of `expect(@solver.factorial(-1)).to eq('Invalid input')` your expectation should be expect(@solver.factorial(-1)).`to raise_exception` 😃